app/views/course/assessment/answer/programming/_test_cases.html.slim

```diff
@@ -13,41 +13,71 @@ table.table.table-striped.table-hover
+      th = Course::Assessment::Question::ProgrammingTestCase.human_attribute_name(:expression)
+      th = Course::Assessment::Question::ProgrammingTestCase.human_attribute_name(:expected)
       th = Course::Assessment::Question::ProgrammingTestCase.human_attribute_name(:hint)
       th = t('.passed')
   tbody
     tr
-      th colspan=4
+      th colspan=5
         = t('.public')
     - public_test_cases.each do |test_case|
       - test_case_result = answer_test_results_hash[test_case]
       = content_tag_for(:tr, test_case) do
         th = format_html(test_case.identifier)
+        td = format_html(test_case.expression)
+        td = format_html(test_case.expected)
         td = format_html(test_case.hint) if !test_case_result || !test_case_result.passed?
         td
           - if test_case_result && test_case_result.passed?
             = fa_icon 'check'.freeze
-          - else
+          - elsif test_case_result
             = fa_icon 'times'.freeze
   - if can?(:grade, answer.answer)
     tbody
       tr
-        th colspan=4
+        th colspan=5
           = t('.private')
       - private_test_cases.each do |test_case|
+        - test_case_result = answer_test_results_hash[test_case]
         = content_tag_for(:tr, test_case) do
           th = format_html(test_case.identifier)
+          td = format_html(test_case.expression)
+          td = format_html(test_case.expected)
+          td = format_html(test_case.hint) if !test_case_result || !test_case_result.passed?
           td
+            - if test_case_result && test_case_result.passed?
+              = fa_icon 'check'.freeze
+            - elsif test_case_result
+              = fa_icon 'times'.freeze
+  - if can?(:grade, answer.answer)
+    tbody
+      tr
+        th colspan=5
+          = t('.evaluation')
+      - evaluation_test_cases.each do |test_case|
+        - test_case_result = answer_test_results_hash[test_case]
+        = content_tag_for(:tr, test_case) do
+          th = format_html(test_case.identifier)
+          td = format_html(test_case.expression)
+          td = format_html(test_case.expected)
+          td = format_html(test_case.hint) if !test_case_result || !test_case_result.passed?
           td
-            - test_case_result = answer_test_results_hash[test_case]
             - if test_case_result && test_case_result.passed?
               = fa_icon 'check'.freeze
-            - else
+            - elsif test_case_result
               = fa_icon 'times'.freeze
 
 - failed_private_test_cases = private_test_cases. \
-    select { |t| !answer_test_results_hash[t] || !answer_test_results_hash[t].passed? }
-- if !failed_private_test_cases.empty? && answer.submission.attempting?
+    select { |t| answer_test_results_hash[t] && !answer_test_results_hash[t].passed? }
+- if !failed_private_test_cases.empty?
     div.panel.panel-danger
-      div.panel-heading = t('.failed_message')
+      div.panel-heading = t('.failed_private_message')
```

Comment:
Think we should not display this to staff when grading ? staff can see the test case details already.